### ISSUE TITLE
Move away from electron

### ISSUE THREAD

**Issue by LadnerJonas** [zulip/zulip-desktop#1240]
**Problem Description**

Electron has serious security flaws and is also using a high amount of memory

**Proposed Solution**

By using https://tauri.app/, which is like electron but written with rust and way better security. More efficient as well

### Comparison between Tauri and Electron

| Detail                     | Tauri  | Electron             |
| -------------------------- | ------ | -------------------- |
| Installer Size Linux       | 3.1 MB | 52.1 MB              |
| Memory Consumption Linux   | 180 MB | 462 MB               |
| Launch Time Linux          | 0.39s  | 0.80s                |
| Interface Service Provider | WRY    | Chromium             |
| Backend Binding            | Rust   | Node.js (ECMAScript) |
| Underlying Engine          | Rust   | V8 (C/C++)           |
| FLOSS                      | Yes    | No                   |
| Multithreading             | Yes    | Yes                  |
| Bytecode Delivery          | Yes    | No                   |
| Multiple Windows           | Yes    | Yes                  |
| Auto Updater               | Yes    | Yes<sup>1</sup>      |
| Custom App Icon            | Yes    | Yes                  |
| Windows Binary             | Yes    | Yes                  |
| macOS Binary               | Yes    | Yes                  |
| Linux Binary               | Yes    | Yes                  |
| iOS Binary                 | Soon   | No                   |
| Android Binary             | Soon   | No                   |
| Desktop Tray               | Yes    | Yes                  |
| Sidecar Binaries           | Yes    | No                   |

**Describe alternatives you've considered**

Native compiled applications

**Comment by andersk** [comment]
Yep, we’ve been following Tauri development with excitement for the last two years, and switching to it is [planned](https://zulip.readthedocs.io/en/latest/contributing/gsoc.html#electron-desktop-app).

**Comment by nothingismagick** [comment]
Amazing. Ping us (the Tauri team) on discord if you run into any issues!
